File: LiveDataExample/LiveDataExampleJava/app/src/main/java/com/tinlone/livedataexamplejava/MyPriceLiveData.java
package com.tinlone.livedataexamplejava;

import androidx.lifecycle.LiveData;


public class MyPriceLiveData extends LiveData<Integer> {

    PriceManager.PriceListener listener = this::setValue;

    @Override
    protected void onActive() {
        PriceManager.getInstance().requestPriceUpdate(listener);
    }

    @Override
    protected void onInactive() {
        PriceManager.getInstance().removeUpdateListener(listener);
    }
}

File: LiveDataExample/LiveDataExampleKotlin/app/src/main/java/com/tinlone/livedataexamplekotlin/MyPriceLiveData.kt
package com.tinlone.livedataexamplekotlin

import androidx.lifecycle.LiveData

class MyPriceLiveData : LiveData<Int?>() {

    var listener: PriceListener = object : PriceListener {
        override fun onPriceChanged(price: Int) {
            value = price
        }
    }

    override fun onActive() {
        super.onActive()
        PriceManager.getInstance().requestPriceUpdate(listener)
    }

    override fun onInactive() {
        super.onInactive()
        PriceManager.getInstance().removeUpdateListener(listener)
    }
}